src/pymovements/gaze/gaze_dataframe.py

```diff
@@ -895,3 +902,41 @@ def _fill_event_detection_kwargs(
             kwargs['timesteps'] = self.frame.get_column('time').to_numpy()
 
         return kwargs
+
+    def _convert_time_units(self, time_unit: str):
+        """Converts the time column to milliseconds based on the specified time unit."""
+        if time_unit is None:
+            raise ValueError(
+                "time_unit must be specified if time_column is specified. "
+                "Supported units are 's' for seconds, 'ms' for milliseconds and "
+                "'step' for steps.",
+            )
+
+        if time_unit == 's':
+            self.frame = self.frame.with_columns(pl.col('time').mul(1000))
+
+        elif time_unit == 'step':
+            if self.experiment is None:
+                raise ValueError("experiment must be specified if time_unit is 'step'")
+
+            self.frame = self.frame.with_columns(
+                pl.col('time').mul(1000).truediv(self.experiment.sampling_rate),
```

Comment:
Sampling rate can be none, add a test for this

Reply:
Actually sampling rate can not be None :)

https://github.com/aeye-lab/pymovements/blob/6bfb14d028a5fb9f0e331c4f6e006523890e3d63/src/pymovements/gaze/experiment.py#L121